### ISSUE TITLE
Authorize with Twitter

### ISSUE THREAD

**Issue by farskid** [hazy-app/webapp#1]
If registrations would have happened through twitter auth, there was no need for having custom usernames as twitter handles could be re-used as usernames. This way sharing would have been easier as well.

For instance everyone who finds me on Twitter, would have known my Hazy link is also `https://hazy.herokuapp.com/[my_twitter_handle]`.

**Comment by nainemom** [comment]
You mean that I force users to auth with twitter? or it just optional register methods next to normal register, facebook auth, etc.

**Comment by farskid** [comment]
It's hard to sync up right now as most people have registered with manual usernames.
If we add it right now even as optional login, still should take care of not available usernames in case someone already used my twitter handle in Hazy.

**Comment by nainemom** [comment]
Yes you right, but what about Instagram, Github, Google, and other socials? For example, my twitter username is @‌‌‌nainemom, but it's not available and taken by another people on Instagram. Is that Instagram fault?

**Comment by farskid** [comment]
It's most likely that this is gonna be shared on twitter heavily. Having that said, it makes sense to either use manual usernames or re-use twitter handles. It's of course my POV and just a suggestion.

**Comment by nainemom** [comment]
One of the biggest my worries about socials auth buttons is that most of them don't have a "just username" scope on the authorization process. They give at least email, gender, name and other basic info that may damage on user trust on Hazy.

**Comment by farskid** [comment]
Fair Enough.

**Comment by nainemom** [comment]
but I still keep this issue open to think about it later. thx :)
